deploy/helpers/devToken.ts

```diff
@@ -0,0 +1,46 @@
+import { BigNumber } from 'ethers';
+import { parseUnits } from 'ethers/lib/utils';
+import { HardhatRuntimeEnvironment } from 'hardhat/types';
+
+async function deployDevToken(
+  env: HardhatRuntimeEnvironment,
+  name: string,
+  contract: string,
+) {
+  const { deployer, alice, bob, carol } = await env.getNamedAccounts();
+  const { deploy, execute, getOrNull, read } = env.deployments;
+
+  const isDeployed = await getOrNull(name);
+
+  if (!isDeployed) {
+    const deployment = await deploy(name, {
+      contract,
+      from: deployer,
+      args: [0],
+      log: true,
+    });
+
+    if (process.env.NODE_ENV !== 'test')
+      await env.tenderly.persistArtifacts({
+        name,
+        address: deployment.address,
+      });
+
+    if (process.env.NODE_ENV !== 'test') {
+      for (let account of [deployer, alice, bob, carol]) {
+        const decimals = await read(name, 'decimals');
+        await execute(
+          name,
+          { from: account },
+          'mint',
+          account,
+          parseUnits('100000', decimals),
+        );
+      }
+    }
+  }
+}
+
+deployDevToken.skip = async (hre: HardhatRuntimeEnvironment) => true;
```

Comment:
Isn't this just a helper function? I'm a bit confused; why did you take it out of its previous place? I don't see it being used anywhere else

Reply:
tbh it's true there is no real need to move it out but imo the devSetup file reads a bit better without it

i was a bit worried that maybe instead of having the if statement that I added to devSetup fixture and running it for both local & ropsten now, that maybe I should add ropstenSetup fixture too and do away with that if statement

in that case having this function in a seperate helper file becomes useful

What are your thoughts on doing it that way?

Reply:
I didn't get all of it. I think the `skip` thing you did here doesn't do anything because this is just a helper function. Overall I think the changes you did are good, but I would move this function back because it's very specific to the dev setup; I'm not sure we want to be re-using the code. Although I do agree that it could be more readable, there's too much nesting on it

Reply:
the skip thing is just so that hardhat doesn't try to execute this function as a deployment script when running deploys

otherwise i'd have to move the function into a file outside of the folder

can do this if needed another time though will move it back where it was

Reply:
moved back